### PR TITLE
Added paragraph about address re-use for BIP131

### PR DESCRIPTION
**Pull request by priestc**
So many people are saying this BIP is bad because it encourages address re-use. I added a paragraph clarifying how address re-use is affected by this BIP.

### CODE REVIEW

File: bip-0131.mediawiki [hunk 1]
```diff
@@ -77,6 +77,26 @@ the user when their wallet contains many UTXOs that qualify it to benefit from
...
 with coalescing transactions in all instances.
 
+== Isn't this BIP bad because it encourage address re-use? ==
+
+Address re-use comes in two forms: re-use by the ''sender'', and re-use by the ''receiver''.
+
+Re-use by the sender is basically using the same address for the change output. This is generally considered bad
+since people looking through your transaction history can determine who you do business with. When
+you generate a new address for every change, your privacy is conserved as it is impossible to know which
+output is a recipient, and which output is the change output. This BIP has '''no effect''' on re-use
```
Comment (MarcoFalke) [inline bip-0131.mediawiki]: > it is impossible to know which output is a recipient

This is not true.

File: bip-0131.mediawiki [hunk 2]
```diff
@@ -77,6 +77,26 @@ the user when their wallet contains many UTXOs that qualify it to benefit from
...
+by the sender.
+
+On the other hand, address re-use by the ''receiver'' occurs under completely different circumstances.
+When you publish an address and have multiple people send to that address, you are engaging in address re-use
+from the receiver. This activity has historically been considered bad because it leads to re-using a private key.
+When you re-use a private key too many times, you run the risk of an attacker performing statistical analysis
+on the multiple signatures, which can lead to an attacker finding out your private key.
```
Comment (sipa) [inline bip-0131.mediawiki]: That's not the primary reason to discourage it, as it's only a theoretical concern. The most important argument against is the same as what you list above: it hurts your own privacy (as now people who send to your will learn something about your incoming funds from others), and by extension, hurts the privacy of everyone interacting with you by enabling more analysis of linkage.

Comment (priestc) [inline bip-0131.mediawiki]: If somebody wants to give up some privacy for a massive decrease in transaction fees, then they should be able to do that. There are many many addresses that have multiple unspent outputs associated with them on the network today, proving that the decrease in privacy is acceptable to many people.

Technically, when you are going through the checkout at Walmart, you can look in to the cash register and get as much information as you would if you looked through someone's bitcoin history through a blockexplorer. Neither Walmart nor any other retailer finds it necessary to keep a privacy guard over the cash register. As long as you rotate your address every so often, its going to take a lot of work to figure out someone's complete financial story by through a block explorer.

By the way, if **I** re-use my address, that has no effect on anyone but myself. Other people are not affected at all.

Comment (sipa) [inline bip-0131.mediawiki]: > By the way, if I re-use my address, that has no effect on anyone but myself. Other people are not affected at all.

I disagree with that statement. You, and you alone, reusing your address simplifies linking other people's addresses together. Due to Bitcoin's lack of full anonymity, privacy provided by the system is a public good, that must be treasured by everyone equally to be preserved.

> If somebody wants to give up some privacy for a massive decrease in transaction fees, then they should be able to do that. 

As a result, I disagree with that as well. We should aim for a design that does not allow one person to get a monetary benefit from hurting the privacy of others in the system.

Comment (priestc) [inline bip-0131.mediawiki]: > You, and you alone, reusing your address simplifies linking other people's addresses together.

If the senders of those transactions were using wallets that generate new addresses for each change output (which I refer to as "non re-use by the sender" in the PR), then their privacy is still preserved, no matter what. Yes, there is a link, but the link is meaningless. When doing analysis, you can't be sure that any output goes to the original owner, or to another recipient. The best you an do is determine a probability that one address is connected to another. If the sender *does* re-use addresses for the change output, then it is a 100% probability that the addresses are linked.

> As a result, I disagree with that as well. We should aim for a design that does not allow one person to get a monetary benefit from hurting the privacy of others in the system.

*Nobody's* privacy is being hurt *anywhere* by enabling wildcard inputs.

**Comment by luke-jr** [comment]
As others have elaborated on, this change is wrong, but GitHub isn't the place to discuss it.

**Pull request merged**